plugins/remove-xlink-ns.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+exports.name = 'remove-xlink-ns';
+exports.type = 'visitor';
+exports.active = false;
+exports.description =
+  'removes xmlns:xlink namespace and replaces xlink:href attributes with href';
+
+/**
+ * removes xmlns:xlink namespace and replaces xlink:href attributes with href
+ *
+ * xlink namespace is obsolete in SVG 2. Xlink:href is recommended to replace with
+ * href attribute.
+ *
+ * https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/href
+ *
+ * @type {import('../lib/types').Plugin<void>}
+ */
+exports.fn = () => {
+  return {
+    element: {
+      enter: (node) => {
+        if (node.attributes['xmlns:xlink'] != null) {
+          delete node.attributes['xmlns:xlink'];
+        }
+        if (node.attributes['xlink:href'] != null) {
+          node.attributes['href'] = node.attributes['xlink:href'];
```

Comment:
I don't think this is always correct. Some older SVG elements don't support `href` and _only_ support `xlink:href`. Mostly elements that were removed in SVG2 but are still supported in engines. Examples include `cursor`, `filter`, `tref` and other SVG fonts related attributes.

You should either have a list of elements this is valid for, or don't do this transform. And therefore, removing `xmlns:xlink` is also not safe when one of those elements that require `xlink:href` is present. However, it should be an option because `xmlns:xlink` is _never_required in HTML.

Reply:
Thanks to both of you for the feedback. I've taken over and cleaned up the PR, so hopefully it should be good to merge.

* [devongovett](https://github.com/devongovett), I've done as you've proposed and made the plugin preserve XLink in that case. It applies for `<cursor>`, `<filter>`, `<font-face-uri>`, `<glyphRef>`, and `<tref>`, with a parameter to force it.
* [yisibl](https://github.com/yisibl), I do not plan to enable the plugin by default myself, but we'll see if any other maintainer would like to chime in.